### PR TITLE
User Profile

### PR DESCRIPTION
**Pull request by Kellswork**
#### What does this PR do?
It adds endpoints that lets users
- edit their profile
- get their profile details
- get their order history

#### How can it be manually tested?
git pull => npm test

#### Are there test specs for this?
YES.

#### Link to the ticket on `Notion`

[user-should-be-able-to-edit-their-profile](https://www.notion.so/kellswork/user-should-be-able-to-edit-their-profile-5d9e4c0932a144fca2771d04763cf73e)
[user-should-be-able-to-view-their-profile](https://www.notion.so/kellswork/user-should-be-able-to-view-their-profile-cb54aa69aa95498ba066cf023b43b047)
[user-should-be-able-to-view-order-history](https://www.notion.so/kellswork/user-should-be-able-to-view-order-history-e714723dcc0b44889022a36bd1467107)

#### Any background contest?
......

### CODE REVIEW

File: src/middlewares/validateUser.js [hunk 1]
```diff
@@ -65,3 +65,50 @@ export const validateLogin = [
     .withMessage('Please input a password'),
   (req, res, next) => handleErrors(req, res, next),
 ];
+
```
Comment (omenkish) [inline src/middlewares/validateUser.js:68]: I am not sure why you did this. We already validated the user, why are you validating profile when profile is just the user details we already have?

File: src/resources/order/order.route.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { verifyAuth, validateAdmin } from '../../middlewares/validateUserAuth';
 import {
   addOrder,
   updateOrder,
-  getOrderById,
+  // getOrderById,
```
Comment (omenkish) [inline src/resources/order/order.route.js:7]: Why are you commenting this out?

File: src/resources/order/order.route.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ import {
   validateStatusUpdate,
   checkStatus,
   checkIfOrderExists,
-  restrictAccessToOwnerAndAdmin,
+  // restrictAccessToOwnerAndAdmin,
```
Comment (omenkish) [inline src/resources/order/order.route.js:16]: This too

**Review by omenkish — CHANGES_REQUESTED** [review]

File: src/middlewares/baseMiddleware.js [hunk 4]
```diff
@@ -15,12 +15,12 @@ export const isAdmin = (user) => user.role.toLowerCase() === 'admin';
 export const userIsOwner = (user, ownerId) => user.id === ownerId;
 
 export const checkUserId = (req, res, next) => {
-  const { id } = req.user;
-  req.body.user_id = id;
+  // const { id } = req.user;
```
[the remark this answers is not in the document]
Comment (Kellswork) [reply src/middlewares/baseMiddleware.js:18]: they weren't useful for the endpoint. I'd remove it

File: src/middlewares/validateUser.js (hunk 1 above)
Comment (Kellswork) [reply src/middlewares/validateUser.js:68]: I didn't want to add confirm-password for when the user changes their password, but I see what you mean

File: src/resources/order/order.route.js (hunk 2 above)
Comment (Kellswork) [reply src/resources/order/order.route.js:7]: mistake

Comment (Kellswork) [reply src/resources/order/order.route.js:7]: I moved the route to users folder and forgot to remove it

File: src/resources/order/order.route.js (hunk 3 above)
Comment (Kellswork) [reply src/resources/order/order.route.js:16]: same thing

File: src/middlewares/baseMiddleware.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Kellswork) [reply src/middlewares/baseMiddleware.js:18]: It was adding the user_id to the req.body which wasn't part of the request body

File: src/middlewares/validateUser.js (hunk 1 above)
Comment (Kellswork) [reply src/middlewares/validateUser.js]: I've edited it to use the user validation

Comment (omenkish) [reply src/middlewares/validateUser.js]: A user still needs to confirm their password

**Review by omenkish — APPROVED** [review]

**Pull request merged**